### PR TITLE
[SMTChecker] Refactor expressions to use SymbolicVariable

### PR DESCRIPTION
**Pull request by leonardoalt**
Depends on #5235

### CODE REVIEW

**Review by chriseth — DISMISSED** [review]
Reviewed the last commit.

**Comment by leonardoalt** [comment]
@chriseth Just rebased after merging the other PR, can you approve again?

**Comment by leonardoalt** [comment]
But don't merge yet, need to check something.

File: libsolidity/formal/SMTChecker.cpp
Comment (leonardoalt) [inline libsolidity/formal/SMTChecker.cpp]: This should be inside `else` otherwise it has no effect.

**Comment by leonardoalt** [comment]
Ok, ready for review/merge again.

**Review by chriseth — APPROVED** [review]

**Pull request merged**
